paymentsheet/src/test/java/com/stripe/android/paymentsheet/forms/FormArgumentsFactoryTest.kt

```diff
@@ -78,6 +81,30 @@ class FormArgumentsFactoryTest {
         assertThat(actualArgs.saveForFutureUseInitialValue).isFalse()
     }
 
+    @Test
+    fun `Create correct FormArguments for null new payment method with setup intent`() {
```

Comment:
It's a little bit hard for me to understand what this test is testing. Is this testing behavior when `supportedAsSavedPaymentMethod` is false? How does the fact that this is using a setup intent affect the result?

I think that this could be clearer with a more specific test name and potentially by asserting the properties of the payment method that are relevant (e.g. if this is testing behavior when `supportedAsSavedPaymentMethod` is false, you could assert that in the test)

Reply:
Yup, super fair. I didn't understand what I was testing when I wrote this 🙈 definitely should have update the name.

I'll figure out something that makes more sense.